### ISSUE TITLE
Programme bloqué avant la génération de la wordlist

### ISSUE THREAD

**Issue by PasteakHacheeUwU** [flegh113/FxWordlist#5]
Salut, après avoir lancé le programme FxWordlist.py je suis bloqué au niveau de la fusion de mes deux wordlists depuis des heures.

### La commmande efféctuée  

> ```
> python fxwordlist.py informations.txt --merge numbers_100_to_99999.txt,weakpass_3w.txt -m optimized
> ```

### La réponse reçue

> ```
> [+] Mode : OPTIMIZED.
> [>] Generation with permutations: 100.00%
> [>] Generation with double letter: 100.00%
> [>] Generation with symbols: 100.00%
> [>] Generation with digits: 100.00%
> [>] Generation with characters addition: 100.00%
> [>] Merger in progress: 100.00%
> ```

Est-ce que c'est normal ? Est-ce que ça peut avoir un rapport avec un fichier corrompu ?

Merci à quiconque me répondera :)

**Comment by flegh113** [comment]
Salut,

Je pense avoir identifié le problème :

Le nombre de combinaisons possibles de mots de passe généré par la fusion des deux Wordlists est probablement extrêmement élevé, ce qui explique le temps de traitement très long pour produire le fichier de sortie.

Dis-toi toujours que si le programme prend énormément de temps pour générer le fichier de sortie, c'est généralement parce que le nombre de combinaisons est trop important.

Si tu as d'autres questions, n'hésite pas à me les poser.

**Comment by flegh113** [comment]
Salut,

Je pense avoir identifié le problème :

Le nombre de combinaisons possibles de mots de passe généré par la fusion des deux wordlists est probablement extrêmement élevé, ce qui explique le temps de traitement très long (voire interminable) pour produire le fichier de sortie.

Dis-toi toujours que si le programme prend énormément de temps pour générer le fichier de sortie, c'est généralement parce que le nombre de combinaisons est trop important, rendant le fichier de sortie inutilisable à cause de la quantité excessive de combinaisons.

Si tu as d'autres questions, n'hésite pas à me les poser.

**Comment by PasteakHacheeUwU** [comment]
Merci beaucoup, il y a donc une possibilité pour que la wordlist ne se génère jamais ?

**Comment by flegh113** [comment]
Effectivement, s'il y a des dizaines et des dizaines voire des centaines de millions de combinaisons de mots de passe, le fichier risque de ne pas être généré. Et même s'il l'était, il serait beaucoup trop volumineux pour être utile.

**Comment by PasteakHacheeUwU** [comment]
Pourtant j'ai une liste fonctionelle de plus d'1 milliard de combinaisons, est-ce qu'il y a une solution que je peux moi-même réaliser pour remédier au problème (à part diminuer le nombre de combinaison) ?

**Comment by flegh113** [comment]
Dans l'idéal, il faudrait optimiser le code au maximum pour qu'il soit le plus performant possible en un minimum de temps. Cela permettrait de générer des fichiers de sortie très volumineux, tout en minimisant les ressources nécessaires.

**Comment by PasteakHacheeUwU** [comment]
Très bien, merci beaucoup d'avoir pris le temps de répondre à mes questions :)
